data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraphData.py

```diff
@@ -0,0 +1,174 @@
+    optional.add_argument('-c','--comments', type=str, default="", metavar='', required=False, help="Comments regarding test details.")
+    optional.add_argument('-fname','--filename', type=str, default="barFigure", metavar='', required=False, help="Name of produced charts.")
+    optional.add_argument('-xstep','--x_step_size', type=int, default=50000, metavar='', required=False, help="Amount of input lines that also determine the step size between tests.")
+    return parser.parse_args()
+
+def makeReducedFile(fileNameOriginal, fileNameReduced, numLines):
+    if Path(fileNameReduced).is_file():
+        return
+    fileReduced = open(fileNameReduced, "w")
+    counter = 0
+    with open(fileNameOriginal) as f:
+        for line in f:
+            counter += 1
+            if counter > numLines:
+                break
+            fileReduced.write(line)
+    fileReduced.close()
+
+def countLinesInFile(filename): ####https://stackoverflow.com/a/27518377/7202012
+    f = open(filename, 'rb')
+    bufgen = takewhile(lambda x: x, (f.raw.read(1024*1024) for _ in repeat(None)))
+    return sum( buf.count(b'\n') for buf in bufgen )
+
+def getTemporaryInDataFileName(number, dataFilename):
+    return dataFilename + str(number) + ".tab"
+
+def getTerminalCommand(args, data):
+    result = args.python_version
+    result += " " + testScriptPath
+    result += " -d " + data
+    if len(args.flags) > 0:
+        result += " -f \"" + args.flags + "\""
+    result += " -r " + str(args.runs)
+    if args.await_tests is True:
+        result += " -a "
+    print("RES: " + result)
+    return result
+
+def generateInputFiles(args):
+    maxNumLines = countLinesInFile(args.data)
+    dataFilename = Path(args.data).stem
+    numDataFiles = 0
+    for i in range(0, int(maxNumLines/minNumLines)):
+        numDataFiles += 1
+        fileNumber = minNumLines*numDataFiles
+        makeReducedFile(args.data, pathToOutputData + "/" + getTemporaryInDataFileName(fileNumber, dataFilename), fileNumber)
+    if numDataFiles == 0:
+        print("Error, data input file must contain at least " + str(minNumLines) + " lines of PSMs.")
+        exit(1)
+    return numDataFiles
+
+def getRunTimes(args):
+    runTimes = []
+    dataFilename = Path(args.data).stem
+    for i in range(1, numDataFiles+1):
+        terminalCommand = getTerminalCommand(args, "{}/{}".format(pathToOutputData, getTemporaryInDataFileName(minNumLines*i, dataFilename)) )
+        p = subprocess.Popen(terminalCommand, stderr=PIPE , stdout=PIPE , text=True, shell=True)
+        print("Running subprocess: " + str(p.pid))
+        p.wait()
+        result = p.stdout.read()
+        print(p.stderr.read())
+        result = result.splitlines()
+        for item in result:
+            if "Min" in item:
+                runTime = re.findall(r"[+]?\d*\.\d+|\d+", item) #This regex script depends on the output from IntegrationTest_Percolator_Speed.py to be in a very specific format.
```

Comment:
Move comment to separate line(s). (This is the kind of comment you particularly don't want to get lost in the right margin.)